cou/steps/charm_operation.py

```diff
@@ -16,16 +16,26 @@
 """Charm operation utilities."""
 import logging
 
-from cou.zaza_utils.model import upgrade_charm
+from cou.utils.juju_utils import async_upgrade_charm
 
 
-def charm_upgrade(application_name: str) -> None:
-    """Upgrade a charm to the latest revision in the current channel."""
+async def charm_upgrade(application_name: str) -> None:
+    """Upgrade a charm to the latest revision in the current channel.
+
+    :param application_name: Name of the Application. E.g: "keystone".
+    :type application_name: str
+    """
     logging.info("Upgrading %s to the latest revision in the current channel", application_name)
-    upgrade_charm(application_name)
+    await async_upgrade_charm(application_name)
+
 
+async def charm_channel_refresh(application_name: str, channel: str) -> None:
+    """Refresh a charm to track a target channel.
 
-def charm_channel_refresh(application_name: str, channel: str) -> None:
-    """Refresh a charm to track a target channel."""
+    :param application_name:  Name of the Application. E.g: keystone.
+    :type application_name: str
+    :param channel: Channel that the application tracks. E.g: "ussuri/stable"
+    :type channel: str
+    """
     logging.info("Refresh %s to the %s channel", application_name, channel)
-    upgrade_charm(application_name, channel=channel)
+    await async_upgrade_charm(application_name, channel=channel)
```

Comment:
why not use libjuju's native upgrade_charm? is it to avoid having to pass a model variable around?

Reply:
It is used just for collecting juju related functions, in case we need juju 3.x compatibility we sould be able to make changes in single file.

Reply:
but `upgrade_charm` has not changed between 2.9 and 3.2